Add more clear_state and change more unit tests to fixtures

mbadams5 opened the pull request.

mbadams5 committed:
Update clear_states and convert many tests to fixture

mbadams5 committed:
Change more tests to use fixture

mbadams5 committed:
Fix various unit tests

Fix unit test issues found using gtest_shuffle

mbadams5 committed:
Try to fix failing unitary system on shuffle

UnitarySystem_VarSpeedCoils is still not passing on shuffle, but
getting closer

mbadams5 committed:
Add more clear states

mbadams5 committed:
Add more clear_states

mbadams5 committed:
Merge branch 'develop' into more_clear_state

# Conflicts:
#	src/EnergyPlus/DataGenerators.cc
#	src/EnergyPlus/DataMoistureBalance.cc
#	src/EnergyPlus/HVACUnitarySystem.cc
#	src/EnergyPlus/PlantChillers.cc
#	src/EnergyPlus/SetPointManager.cc
#	src/EnergyPlus/WindowComplexManager.cc
#	src/EnergyPlus/WindowEquivalentLayer.cc

mbadams5 committed:
Fix merge conflicts

# Conflicts:
#	src/EnergyPlus/DataGenerators.cc
#	src/EnergyPlus/DataMoistureBalance.cc
#	src/EnergyPlus/HVACUnitarySystem.cc
#	src/EnergyPlus/PlantChillers.cc
#	src/EnergyPlus/SetPointManager.cc
#	src/EnergyPlus/WindowComplexManager.cc
#	src/EnergyPlus/WindowEquivalentLayer.cc

mbadams5 committed:
Add more clear_state and update unit tests

mbadams5 committed:
Merge branch 'develop' into more_clear_state

# Conflicts:
#	src/EnergyPlus/DataRoomAirModel.hh
#	src/EnergyPlus/OutdoorAirUnit.cc
#	src/EnergyPlus/RoomAirModelAirflowNetwork.cc
#	tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc
#	tst/EnergyPlus/unit/LowTempRadiantSystem.unit.cc
#	tst/EnergyPlus/unit/RoomAirflowNetwork.unit.cc

mbadams5 committed:
Fix merge conflicts

# Conflicts:
#	src/EnergyPlus/DataRoomAirModel.hh
#	src/EnergyPlus/OutdoorAirUnit.cc
#	src/EnergyPlus/RoomAirModelAirflowNetwork.cc
#	tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc
#	tst/EnergyPlus/unit/LowTempRadiantSystem.unit.cc
#	tst/EnergyPlus/unit/RoomAirflowNetwork.unit.cc

Myoldmopar commented:
@mbadams5 I'm building and running locally to figure out what the diffs are...

Myoldmopar commented:
Hmmmm yeah, the eso diffs are all super tiny.  10^-5 Joules, 10^-11 degC  I don't have an explanation for them right now, but they are super tiny and very much repeatable.  Maybe consider pulling out some of your changes that you feel could potentially be suspicious, and I'll re-run the diffs locally for a quick result.

mbadams5 committed:
Enable unit test shuffle by default

This will help to bring order dependency issues with the unit tests to
light sooner.

mbadams5 committed:
Add back necessary (for now) statics

mbadams5 committed:
Add statics into namespace

This fixes a failing (shuffled) unit test and why I removed the statics
in the first place.

Myoldmopar commented:
Uuuuuuuuuummmmm https://github.com/NREL/EnergyPlus/pull/5391#commitcomment-15273551

Myoldmopar commented:
@mbadams5 The Mac CI had an issue, so I'm deleting those results and will let it retry.

Myoldmopar commented:
But I am getting no diffs just like the Ubuntu build results show, so this should be good to go in here shortly.
